Source/RestClient.swift

```diff
@@ -0,0 +1,133 @@
+                callback?(self.token, nil)
+            } else {
+                callback?(nil, error)
+            }
+        }
+    }
+
+    open func refresh(callback: ((_ token: TokenInfo?, _ error: HTTPError?) -> Void)? = nil) {
+        if let token = token {
+            let parameters: [String: String] = [
+                "grant_type": "refresh_token",
+                "refresh_token": token.refresh_token!,
+                "endpoint_id": token.endpoint_id!
+            ]
+            let headers: [String: String] = ["Authorization": basicAuthToken()]
+            postString("/restapi/oauth/token", parameters: parameters, encoding: URLEncoding.default, headers: headers) { string, error in
+                if error == nil {
+                    self.token = TokenInfo(JSONString: string!)
+                    callback?(self.token, nil)
+                } else {
+                    callback?(nil, error)
+                }
+            }
+        }
+    }
+
+    open func revoke(callback: ((Bool) -> Void)? = nil) {
+        if let token = token {
+            let parameters: [String: String] = ["token": token.access_token!]
+            let headers: [String: String] = ["Authorization": basicAuthToken()]
+            postString("/restapi/oauth/revoke", parameters: parameters, encoding: URLEncoding.default, headers: headers) { string, error in
+                callback?(error == nil)
+            }
+        } else {
+            callback?(true)
+        }
+    }
+
+    open func authorizeUri(redirectUri: String, state: String = "") -> String {
+        var urlComponents = URLComponents(string: server)!
+        urlComponents.path = "/restapi/oauth/authorize"
+        urlComponents.queryItems = [
+            URLQueryItem(name: "response_type", value: "code"),
+            URLQueryItem(name: "state", value: state),
+            URLQueryItem(name: "redirect_uri", value: redirectUri),
+            URLQueryItem(name: "client_id", value: appKey),
+        ]
+        return urlComponents.string!
+    }
+
+    open func authorize(authCode: String, redirectUri: String, callback: ((_ token: TokenInfo?, _ error: HTTPError?) -> Void)? = nil) {
+        let parameters: [String: String] = [
+            "grant_type": "authorization_code",
+            "redirect_uri": redirectUri,
+            "code": authCode,
+        ]
+        let headers: [String: String] = ["Authorization": basicAuthToken()]
+        postString("/restapi/oauth/token", parameters: parameters, encoding: URLEncoding.default, headers: headers) { string, error in
+            if error == nil {
+                self.token = TokenInfo(JSONString: string!)
+                callback?(self.token, nil)
+            } else {
+                callback?(nil, error)
+            }
+        }
+    }
+
+    open func restapi(_ _id: String? = "v1.0") -> RestapiPath {
+        return RestapiPath(parent: PathSegment(parent: nil, _id: nil, rc: self), _id: _id)
+    }
+
+}
```

Comment:
The comment was about RestClient in general. Please make sure your classes are following reference, SDK (e.g. Http/Subscriptions) part is clearly separated from Client part.